orca-queue/src/main/kotlin/com/netflix/spinnaker/config/QueueConfiguration.kt

```diff
@@ -61,6 +61,8 @@ open class QueueConfiguration {
       ThreadPoolTaskExecutor().apply {
         corePoolSize = 20
         maxPoolSize = 150
+        setQueueCapacity(20)
```

Comment:
Chose that number sort of arbitrarily, I suppose keeping a work queue at all is what's causing processing anomalies - now that you mention that, fixed thread pool is probably the best way forward.